Modules/include/VersionInfoProvider.h

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include "ApplicationModule.h"
+#include "ConfigReader.h"
+#include "ScalarAccessor.h"
+#include "VersionInfo.h"
+#include "boost/format.hpp"
+
+namespace ChimeraTK::VersionInfo {
+
+  extern const int major;
+  extern const int minor;
+  extern const int applicationPatch;
+
+} // namespace ChimeraTK::VersionInfo
+
+namespace ChimeraTK {
+  /**
+   * This module can be added to applications to provide version information from the `CMakeLists.txt` file in a
+   * standardized manner.
+   * The server is also expected to have a module named "Application" with a variable named "configPatchVersion" of
+   * type "int32" in its "config.xml" file.
+   */
+  class VersionInfoProvider : public ApplicationModule {
+   public:
+    explicit VersionInfoProvider(Application* owner)
+    : ApplicationModule(owner, "Application", "Module to publish version information") {
```

Comment:
please move the implementation down below the class definition, separated by a "chrome bar" (full line length comment `/*******/`)